[Bug]: Recursive update in ConcurrentHashMap

### Your Environment

```stacktarce
Cannot create listener

com.intellij.diagnostic.PluginException: Cannot create class io.unthrottled.doki.icons.jetbrains.shared.listeners.ApplicationLifecycleListener (classloader=PluginClassLoader(plugin=PluginDescriptor(name=Doki Theme Icons, id=io.unthrottled.doki.icons, descriptorPath=plugin.xml, path=D:\JetBrains\Toolbox\apps\IDEA-C\ch-0\223.8214.27.plugins\Doki Theme Icons, version=88.3-1.0.2, package=null, isBundled=false), packagePrefix=null, instanceId=20, state=active))
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:989)
	at com.intellij.serviceContainer.ComponentManagerImpl.createListener(ComponentManagerImpl.kt:1027)
	at com.intellij.util.messages.impl.CompositeMessageBus.subscribeLazyListeners$lambda$3(CompositeMessageBus.kt:118)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeNonCancelableSection$3(CoreProgressManager.java:224)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:664)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:620)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$computeInNonCancelableSection$4(CoreProgressManager.java:232)
	at com.intellij.openapi.progress.Cancellation.computeInNonCancelableSection(Cancellation.java:99)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeInNonCancelableSection(CoreProgressManager.java:232)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeNonCancelableSection(CoreProgressManager.java:223)
	at com.intellij.util.messages.impl.CompositeMessageBus.subscribeLazyListeners(CompositeMessageBus.kt:112)
	at com.intellij.util.messages.impl.CompositeMessageBus.doComputeSubscribers(CompositeMessageBus.kt:93)
	at com.intellij.util.messages.impl.MessageBusImpl.computeSubscribers(MessageBusImpl.kt:192)
	at com.intellij.util.messages.impl.CompositeMessageBus.computeSubscribers(CompositeMessageBus.kt:88)
	at com.intellij.util.messages.impl.ToDirectChildrenMessagePublisher$publish$handlers$1.invoke(CompositeMessageBus.kt:260)
	at com.intellij.util.messages.impl.ToDirectChildrenMessagePublisher$publish$handlers$1.invoke(CompositeMessageBus.kt:260)
	at com.intellij.util.messages.impl.ToDirectChildrenMessagePublisher.publish$lambda$0(CompositeMessageBus.kt:260)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
	at com.intellij.util.messages.impl.ToDirectChildrenMessagePublisher.publish$intellij_platform_core(CompositeMessageBus.kt:260)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:430)
	at jdk.proxy2/jdk.proxy2.$Proxy95.appFrameCreated(Unknown Source)
	at com.intellij.idea.IdeStarter.openProjectIfNeeded$suspendImpl(IdeStarter.kt:91)
	at com.intellij.idea.IdeStarter.openProjectIfNeeded(IdeStarter.kt)
	at com.intellij.idea.IdeStarter$start$2.invokeSuspend(IdeStarter.kt:72)
	at com.intellij.idea.IdeStarter$start$2.invoke(IdeStarter.kt)
	at com.intellij.idea.IdeStarter$start$2.invoke(IdeStarter.kt)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:89)
	at kotlinx.coroutines.CoroutineScopeKt.coroutineScope(CoroutineScope.kt:264)
	at com.intellij.idea.IdeStarter.start$suspendImpl(IdeStarter.kt:70)
	at com.intellij.idea.IdeStarter.start(IdeStarter.kt)
	at com.intellij.idea.ApplicationLoader.initApplicationImpl(ApplicationLoader.kt:169)
	at com.intellij.idea.ApplicationLoader.access$initApplicationImpl(ApplicationLoader.kt:1)
	at com.intellij.idea.ApplicationLoader$initApplicationImpl$1.invokeSuspend(ApplicationLoader.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
Caused by: java.lang.ExceptionInInitializerError
	at io.unthrottled.doki.icons.jetbrains.shared.listeners.ApplicationLifecycleListener.<clinit>(ApplicationLifecycleListener.kt:12)
	at java.base/jdk.internal.misc.Unsafe.allocateInstance(Native Method)
	at java.base/java.lang.invoke.DirectMethodHandle.allocateInstance(DirectMethodHandle.java:520)
	at com.intellij.serviceContainer.ComponentManagerImpl.instantiateClass(ComponentManagerImpl.kt:950)
	... 38 more
Caused by: java.lang.IllegalStateException: Recursive update
	at java.base/java.util.concurrent.ConcurrentHashMap.replaceNode(ConcurrentHashMap.java:1167)
	at java.base/java.util.concurrent.ConcurrentHashMap.remove(ConcurrentHashMap.java:1102)
	at com.intellij.util.messages.impl.MessageBusImpl.notifyOnSubscription(MessageBusImpl.kt:199)
	at com.intellij.util.messages.impl.BaseBusConnection.subscribe(BaseBusConnection.kt:41)
	at io.unthrottled.doki.icons.jetbrains.shared.path.IconPathReplacementComponent.initialize(IconPathReplacementComponent.kt:51)
	at io.unthrottled.doki.icons.jetbrains.shared.integrations.PlatformHacker.<clinit>(PlatformHacker.kt:15)
	... 42 more

```

```
IntelliJ IDEA with Maho 2022.3.1 RC (Community Edition)
Build #IC-223.8214.27, built on December 15, 2022
Runtime version: 17.0.4+8-LTS amd64
VM: OpenJDK 64-Bit Server VM by SAP SE
Windows 10 10.0
GC: Shenandoah Pauses, Shenandoah Cycles
Memory: 16384M
Cores: 32
Registry:
    ide.intellij.laf.enable.animation=true
    debugger.watches.in.variables=false
    ide.tooltip.initialDelay=0
    ide.balloon.shadow.size=0

Non-Bundled Plugins:
    io.acari.DDLCTheme (88.3-1.8.0)
    amadeus.maho (223.8214.27)
    io.unthrottled.doki.icons (88.3-1.0.2)

Kotlin: 223-1.7.21-release-272-IJ8214.27
```

### How to reproduce.

startup intellij

### What is expected?

no exception occurred

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,11 @@
 
 ## [Unreleased]
 
+## 88.3-1.0.3 [Start Up Fix]
+
+### Fixed
+- Issue on plugin startup for IDEs with non-JetBrains java runtimes.
+
 ## 88.3-1.0.2 [Platform SDK Migration]
 
 ### Removed

--- a/RELEASE-NOTES.md
+++ b/RELEASE-NOTES.md
@@ -1,8 +1,3 @@
-# 88.3-1.0.1 [Initial Release]
+### Fixed
 
-<div align="center">
-<img src="https://doki.assets.unthrottled.io/misc/doki_icons_logo.png" width="256" height="256" alt="Doki Theme Icons"></img>
-</div>
-
-
-The first public release of the plugin!
+- Issue on plugin startup for IDEs with non-JetBrains java runtimes.

--- a/gradle.properties
+++ b/gradle.properties
@@ -2,7 +2,7 @@
 
 pluginGroup = io.unthrottled.doki.icons
 pluginName = Doki Theme Icons
-pluginVersion = 88.3-1.0.2
+pluginVersion = 88.3-1.0.3
 
 # Supported build number ranges and IntelliJ Platform versions -> https://plugins.jetbrains.com/docs/intellij/build-number-ranges.html
 pluginSinceBuild = 222

--- a/shared/src/main/kotlin/io/unthrottled/doki/icons/jetbrains/shared/integrations/PlatformHacker.kt
+++ b/shared/src/main/kotlin/io/unthrottled/doki/icons/jetbrains/shared/integrations/PlatformHacker.kt
@@ -12,7 +12,7 @@ import javassist.expr.MethodCall
 object PlatformHacker : Logging {
 
   init {
-    IconPathReplacementComponent.initialize()
+    IconPathReplacementComponent.installComponents()
     hackPlatform()
   }
 

--- a/shared/src/main/kotlin/io/unthrottled/doki/icons/jetbrains/shared/onboarding/UpdateNotification.kt
+++ b/shared/src/main/kotlin/io/unthrottled/doki/icons/jetbrains/shared/onboarding/UpdateNotification.kt
@@ -20,7 +20,7 @@ private fun buildUpdateMessage(): String =
   """
       What's New?<br>
       <ul>
-        <li>Initial Release!</li>
+        <li>Fixed a startup issue.</li>
       </ul>
       <br>See the <a href="https://github.com/doki-theme/doki-theme-icons-jetbrains#documentation">documentation</a> for features, usages, and configurations.
       <br>The <a href="https://github.com/doki-theme/doki-theme-icons-jetbrains/blob/master/CHANGELOG.md">changelog</a> is available for more details.

--- a/shared/src/main/kotlin/io/unthrottled/doki/icons/jetbrains/shared/path/IconPathReplacementComponent.kt
+++ b/shared/src/main/kotlin/io/unthrottled/doki/icons/jetbrains/shared/path/IconPathReplacementComponent.kt
@@ -47,8 +47,12 @@ object IconPathReplacementComponent : IconConfigListener {
   private val connection = ApplicationManager.getApplication().messageBus.connect()
 
   fun initialize() {
-    ExperimentalUIBastardizer.bastardizeExperimentalUI()
     this.connection.subscribe(IconConfigListener.TOPIC, this)
+    installComponents()
+  }
+
+  fun installComponents() {
+    ExperimentalUIBastardizer.bastardizeExperimentalUI()
 
     iconInstallPacs.forEach { pak ->
       IconLoader.removePathPatcher(pak.iconPatcher)

